File: 1402-reducing-dishes/1402-reducing-dishes.py
class Solution:
    def maxSatisfaction(self, s: List[int]) -> int:
        s.sort(reverse=True)
        if(s[0]<0):
            return 0
        a=m=s[0]
        for i in range(1, len(s)):
            a+=s[i]
            if m+a<m :
                return m
            m+=a
        return m

File: 1402-reducing-dishes/1402-reducing-dishes.java
import java.util.Arrays;
import java.util.Collections;
class Solution {
    public int maxSatisfaction(int[] sa) {
        Integer[] s=Arrays.stream(sa).boxed().toArray(Integer[] ::new);
        Arrays.sort(s, Collections.reverseOrder());
        if(s[0]<0){
            return 0;
        }
        int m=s[0];
        int p=s[0];
        for(int i=1;i<s.length;i++){
            p+=s[i];
            if(m+p<m){
                return m;
            }
            m+=p;
        }
        return m;
    }
}